ntbs-service/DataMigration/NotificationMapper.cs

```diff
@@ -221,22 +224,34 @@ private async Task<IEnumerable<IList<Notification>>> GetGroupedResultsAsNotifica
             IEnumerable<TreatmentEvent> notificationOutcomeEvents)
         {
             return notification.ClinicalDetails.IsPostMortem == true
-                ? TreatmentEventsForPostMortemNotification(rawNotification)
+                ? TreatmentEventsForPostMortemNotification(rawNotification, notificationOutcomeEvents)
                 : TreatmentEventsForPreMortemNotification(
                     notification,
                     rawNotification,
                     notificationTransferEvents,
                     notificationOutcomeEvents);
         }
 
-        private List<TreatmentEvent> TreatmentEventsForPostMortemNotification(MigrationDbNotification rawNotification)
+        private List<TreatmentEvent> TreatmentEventsForPostMortemNotification(MigrationDbNotification rawNotification,
+            IEnumerable<TreatmentEvent> notificationOutcomeEvents)
         {
             // For post mortem cases the death event is the ONLY event we want to import so the final outcome is
             // correctly reported.
-            return new List<TreatmentEvent>
+            var deathEvents = notificationOutcomeEvents.Where(e => _diedOutcomeIds.Contains(e.TreatmentOutcomeId ?? 0)).ToList();
+            TreatmentEvent eventToReturn;
+            if (deathEvents.Any())
+            {
+                eventToReturn = deathEvents.OrderByDescending(e => e.EventDate).First();
+                eventToReturn.EventDate = rawNotification.DeathDate;
+                eventToReturn.TreatmentOutcome =
+                    _diedOutcomes.Single(te => te.TreatmentOutcomeId == eventToReturn.TreatmentOutcomeId);
```

Comment:
Why isn't the treatment outcome already on the event?

Reply:
I think because we have only set the outcome ID, and it hasn't gone to the DB to grab which outcome that corresponds to

Reply:
I guess my question is really: given that we don't do this in the other path (non post-mortem) where we just append the events to the notification, why do we need to do it in the post-mortem case?

Reply:
The commit which added the adding of an outcome was to fix this bug: https://airelogic-nis.atlassian.net/browse/NTBS-1650?focusedCommentId=15258
I'm not 100% sure why we don't do it for other outcomes.

Reply:
Okay, it looks like I may have solved the issue this was trying to overcome in NTBS-2478, when we no longer relied on loaded treatment outcomes to check "should be closed". Still looking, but that looks likely.

Reply:
Okay I've cracked the case, it's so that `HasDeathEventForPostMortemCase` has a loaded outcome. 
2 approaches we could take:
- no longer rely on the loaded outcome, we can easily pull it from SeedData as we do in `Notification.ShouldBeClosed`
- keep as is, load the outcomes like I've done above
I personally think we should just be getting them from seed data, which isn't in the database so will be more performant anyway. I think we should also change the `_postMortemType` field to also get it's values from SeedData, as there's no point going to the DB for something we have. @JoshCadney agree, or shall we stick with what I have here?

Reply:
I think at this stage we should probably leave it alone, because it's the safer thing to do.

We could use the seed data for `HasDeathEventForPostMortemCase` and that would be pretty safe, however, changing `_postMortemOutcomeType` is more risky - it may be that we need it to be the same object that is in the entity framework context, rather than just an object containing the same data.

Reply:
Yeah okay, I will leave it as is for now. I think Benji created a ticket to deal with how we get `_postMortemOutcomeType` when this was last looked at so I might add something to that ticket about it. I'm going to add a comment about why we are loading the outcomes for post mortem cases because it isn't clear and someone could easily go down this rabbit hole again

Reply:
Yes, that's a good idea :-)